zanata-war/src/main/webapp/iteration/view.xhtml

```diff
@@ -84,16 +84,31 @@
     return false;
   }
 
-  function checkIfSpecificFileSelected() {
-    checkIfFileSelected(#{rich:element('specificDocFileInput')},
-      #{rich:element('specificDocSubmitUploadButton')});
+  function checkIfFileSelected() {
+    var isFileSelected = jQuery(getSourceDocInput()).val() != '';
+    toggleUploadSourceButton(isFileSelected);
+    if(isFileSelected) {
+      renderSourceDocTypeSelector(jQuery(getSourceDocInput()).val());
+    }
   }
 
-  function checkIfFileSelected(fileInput, submitButton) {
-    var noFileSelected = fileInput.value === undefined ||
-      fileInput.value === "";
-    submitButton.disabled = noFileSelected;
+  function getSourceDocTypeSelection() {
+    return jQuery('#' + jqSelector('#{rich:clientId('sourceDocumentTypes')}'));
   }
+
+  function getSourceDocInput() {
+    return jQuery('#' + jqSelector('#{rich:clientId('specificDocFileInput')}'));
+  }
+
+  function getUploadSourceDocButton() {
+    return jQuery('#' + jqSelector('#{rich:clientId('specificDocSubmitUploadButton')}'));
+  }
+
+  function toggleUploadSourceButton(enable) {
```

Comment:
- [ ] suggest `setUploadSourceButtonEnabled(boolean)` so that it is clear what it does.